Update cypress, initial volto-testing package (#3369)
plone/volto: docs/source/upgrade-guide/index.md

Please update your projects to a NodeJS LTS version, where either 14 or 16 is supported at the moment of this writing.
Version 16 is recommended.

### Removed `date-fns` from build

The `date-fns` library has been removed from Volto's dependencies.
It was in the build because `Cypress` depended on it.

> After the `Cypress` upgrade it no longer depends on it.

Reviewer: ```suggestion
After `Cypress` was upgraded, it no longer depends on `date-fns`.
```

> If your project still depends on it, add it as a dependency of your project.

Reviewer: ```suggestion
If your project still depends on `date-fns`, add it as a dependency of your project.
```

```{warning}

> The `date-fns` version present in the build was quite an old one (1.x.x series). Beware to use an updated version (2.x.x), since it may contain quite some breaking changes. Also take a look at: https://dockyard.com/blog/2020/02/14/you-probably-don-t-need-moment-js-anymore

Reviewer: ```suggestion
The `date-fns` version present in the build was quite old (1.x.x series).
Beware when using an updated version (2.x.x), as it may contain some breaking changes.
Also take a look at: https://dockyard.com/blog/2020/02/14/you-probably-don-t-need-moment-js-anymore
```

> and the new `FormattedDate` component in Volto core.

Reviewer: This is a sentence fragment. I'm not sure what is the intent. Please remove or enhance it.

Author: Done, how about now?

The `date-fns` library has been removed from Volto's dependencies.
It was in the build because `Cypress` depended on it.
After `Cypress` was upgraded, it no longer depends on `date-fns`.
If your project still depends on `date-fns`, add it as a dependency of your project.

```{warning}
The `date-fns` version present in the build was quite old (1.x.x series).
Beware when using an updated version (2.x.x), as it may contain some breaking changes.
Also take a look at: https://dockyard.com/blog/2020/02/14/you-probably-don-t-need-moment-js-anymore

> If you need to format dates in Volto it's recommended to use the `FormattedDate` component present already in Volto core.

Reviewer: ```suggestion
If you need to format dates in Volto, it's recommended to use the `FormattedDate` component in Volto core.
```

> It already uses the modern recommendations for date formatting in the web.

Reviewer: ```suggestion
It uses modern recommendations for date formatting on the web.
```
